perf/impl/go-libp2p/v0.27/perf.go

```diff
@@ -0,0 +1,126 @@
+package main
+
+import (
+	"context"
+	"encoding/binary"
+	"fmt"
+	"io"
+	"time"
+
+	logging "github.com/ipfs/go-log/v2"
+	pool "github.com/libp2p/go-buffer-pool"
+	"github.com/libp2p/go-libp2p/core/host"
+	"github.com/libp2p/go-libp2p/core/network"
+	"github.com/libp2p/go-libp2p/core/peer"
+)
+
+var log = logging.Logger("perf")
+
+const (
+	BlockSize = 64 << 10
+
+	ID = "/perf/1.0.0"
+)
+
+type PerfService struct {
+	Host host.Host
+}
+
+func NewPerfService(h host.Host) *PerfService {
+	ps := &PerfService{h}
+	h.SetStreamHandler(ID, ps.PerfHandler)
+	return ps
+}
+
+func (p *PerfService) PerfHandler(s network.Stream) {
+	u64Buf := make([]byte, 8)
+	_, err := io.ReadFull(s, u64Buf)
+	if err != nil {
+		log.Errorw("err", err)
+		s.Reset()
+		return
+	}
+
+	bytesToSend := binary.BigEndian.Uint64(u64Buf)
+
+	_, err = p.drainStream(context.Background(), s)
+	if err != nil {
+		log.Errorw("err", err)
+		s.Reset()
+		return
+	}
+
+	err = p.sendBytes(context.Background(), s, bytesToSend)
+	if err != nil {
+		log.Errorw("err", err)
+		s.Reset()
+		return
+	}
+
+}
+
+func (ps *PerfService) sendBytes(ctx context.Context, s network.Stream, bytesToSend uint64) error {
+	buf := pool.Get(BlockSize)
```

Comment:
nit: You can stack allocate this buf in the `RunPerf` function and use it instead of needing a sync.Pool.

Reply:
I don't think it matters. It would even be fine allocating this in this function directly. This is a single 64k allocation per run of the perf protocol. Clarity over micro-optimization.